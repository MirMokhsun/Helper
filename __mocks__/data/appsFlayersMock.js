export const dataForAppsFlyer = {
    "campaign": 'value_campaign',
    "af_sub1": 'value_af_sub1',
    "af_sub2": 'value_af_sub2',
    "af_sub3": 'value_af_sub3',
    "af_adset": 'value_af_adset',
}