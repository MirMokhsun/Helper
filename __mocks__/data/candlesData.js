export const dataCandles = [
    { "T": 1, "O": 1.0823, "C": 1.0807, "L": 1.0795, "H": 1.0826 },
    { "T": 2, "O": 1.0808, "C": 1.0797, "L": 1.0773, "H": 1.0808 },
    { "T": 3, "O": 1.0796, "C": 1.0805, "L": 1.0789, "H": 1.0808 },
    { "T": 4, "O": 1.0804, "C": 1.0798, "L": 1.0796, "H": 1.0809 },
    { "T": 5, "O": 1.0798, "C": 1.0798, "L": 1.0795, "H": 1.0801 },
    { "T": 6, "O": 1.0798, "C": 1.0792, "L": 1.0779, "H": 1.0798 },
    { "T": 7, "O": 1.0792, "C": 1.0791, "L": 1.0781, "H": 1.08 },
    { "T": 8, "O": 1.079, "C": 1.0783, "L": 1.0782, "H": 1.0806 },
    { "T": 9, "O": 1.0788, "C": 1.0734, "L": 1.0728, "H": 1.0793 },
    { "T": 10, "O": 1.0734, "C": 1.0758, "L": 1.0734, "H": 1.076 },
    { "T": 11, "O": 1.0757, "C": 1.0762, "L": 1.0754, "H": 1.0765 },
    { "T": 12, "O": 1.0763, "C": 1.0785, "L": 1.0759, "H": 1.0789 },
    { "T": 13, "O": 1.0785, "C": 1.0764, "L": 1.0761, "H": 1.0785 },
    { "T": 14, "O": 1.0764, "C": 1.0768, "L": 1.076, "H": 1.0785 },
    { "T": 15, "O": 1.0768, "C": 1.0713, "L": 1.0713, "H": 1.0768 },
    { "T": 16, "O": 1.0714, "C": 1.0723, "L": 1.0709, "H": 1.0728 },
    { "T": 17, "O": 1.0722, "C": 1.0722, "L": 1.0712, "H": 1.0726 },
    { "T": 18, "O": 1.0722, "C": 1.0749, "L": 1.0718, "H": 1.0754 },
    { "T": 19, "O": 1.0747, "C": 1.0745, "L": 1.0737, "H": 1.0759 },
    { "T": 20, "O": 1.0745, "C": 1.0724, "L": 1.0723, "H": 1.0746 },
    { "T": 21, "O": 1.0726, "C": 1.0735, "L": 1.0724, "H": 1.074 },
    { "T": 22, "O": 1.0734, "C": 1.0737, "L": 1.0732, "H": 1.0745 },
    { "T": 23, "O": 1.0736, "C": 1.0745, "L": 1.0736, "H": 1.0745 },
    { "T": 24, "O": 1.0745, "C": 1.0745, "L": 1.0741, "H": 1.0751 },
    { "T": 25, "O": 1.0745, "C": 1.0762, "L": 1.0745, "H": 1.0772 },
    { "T": 26, "O": 1.0762, "C": 1.0756, "L": 1.0756, "H": 1.0771 },
    { "T": 27, "O": 1.0757, "C": 1.0749, "L": 1.0746, "H": 1.0761 },
    { "T": 28, "O": 1.0749, "C": 1.0753, "L": 1.0749, "H": 1.0755 },
    { "T": 29, "O": 1.0753, "C": 1.0749, "L": 1.0745, "H": 1.0755 },
    { "T": 30, "O": 1.0749, "C": 1.0748, "L": 1.0748, "H": 1.0754 },
    { "T": 31, "O": 1.0749, "C": 1.0751, "L": 1.0747, "H": 1.0754 },
    { "T": 32, "O": 1.0751, "C": 1.0775, "L": 1.0747, "H": 1.0781 },
    { "T": 33, "O": 1.0773, "C": 1.0804, "L": 1.077, "H": 1.0817 },
    { "T": 34, "O": 1.0804, "C": 1.0798, "L": 1.0785, "H": 1.0812 },
    { "T": 35, "O": 1.08, "C": 1.0781, "L": 1.0779, "H": 1.0804 },
    { "T": 36, "O": 1.0781, "C": 1.0766, "L": 1.0766, "H": 1.0781 },
    { "T": 37, "O": 1.0766, "C": 1.0768, "L": 1.0758, "H": 1.0773 },
    { "T": 38, "O": 1.0769, "C": 1.0734, "L": 1.0728, "H": 1.077 },
    { "T": 39, "O": 1.0735, "C": 1.0727, "L": 1.0721, "H": 1.0743 },
    { "T": 40, "O": 1.0726, "C": 1.0728, "L": 1.0723, "H": 1.0751 },
    { "T": 41, "O": 1.0728, "C": 1.0726, "L": 1.072, "H": 1.0741 },
    { "T": 42, "O": 1.0724, "C": 1.0732, "L": 1.0719, "H": 1.0732 },
    { "T": 43, "O": 1.0732, "C": 1.0716, "L": 1.0714, "H": 1.0737 },
    { "T": 44, "O": 1.0717, "C": 1.0716, "L": 1.0715, "H": 1.0726 },
    { "T": 45, "O": 1.0716, "C": 1.0729, "L": 1.0715, "H": 1.073 },
    { "T": 46, "O": 1.073, "C": 1.0722, "L": 1.0721, "H": 1.0733 },
    { "T": 47, "O": 1.0724, "C": 1.0724, "L": 1.0718, "H": 1.073 },
    { "T": 48, "O": 1.0723, "C": 1.0725, "L": 1.0719, "H": 1.0728 },
    { "T": 49, "O": 1.0724, "C": 1.0738, "L": 1.072, "H": 1.0741 },
    { "T": 50, "O": 1.0739, "C": 1.0739, "L": 1.0737, "H": 1.075 },
    { "T": 51, "O": 1.0739, "C": 1.0756, "L": 1.0737, "H": 1.076 },
    { "T": 52, "O": 1.0757, "C": 1.0748, "L": 1.0748, "H": 1.076 },
    { "T": 53, "O": 1.0749, "C": 1.0741, "L": 1.0737, "H": 1.0749 },
    { "T": 54, "O": 1.0741, "C": 1.0747, "L": 1.0741, "H": 1.0749 },
    { "T": 55, "O": 1.0747, "C": 1.0753, "L": 1.0746, "H": 1.0755 },
    { "T": 56, "O": 1.0751, "C": 1.0706, "L": 1.0701, "H": 1.0757 },
    { "T": 57, "O": 1.0707, "C": 1.0716, "L": 1.0703, "H": 1.0725 },
    { "T": 58, "O": 1.0716, "C": 1.0715, "L": 1.0714, "H": 1.073 },
    { "T": 59, "O": 1.0714, "C": 1.0711, "L": 1.0697, "H": 1.0714 },
    { "T": 60, "O": 1.0711, "C": 1.0696, "L": 1.0695, "H": 1.0711 },
    { "T": 61, "O": 1.0696, "C": 1.0699, "L": 1.0688, "H": 1.0713 },
    { "T": 62, "O": 1.0699, "C": 1.0712, "L": 1.0694, "H": 1.0722 },
    { "T": 63, "O": 1.0713, "C": 1.0701, "L": 1.0696, "H": 1.0718 },
    { "T": 64, "O": 1.0701, "C": 1.0672, "L": 1.0671, "H": 1.0715 },
    { "T": 65, "O": 1.0673, "C": 1.0699, "L": 1.0666, "H": 1.07 },
    { "T": 66, "O": 1.07, "C": 1.0696, "L": 1.0694, "H": 1.0715 },
    { "T": 67, "O": 1.0695, "C": 1.0672, "L": 1.067, "H": 1.0697 },
    { "T": 68, "O": 1.0672, "C": 1.0681, "L": 1.0667, "H": 1.0683 },
    { "T": 69, "O": 1.0682, "C": 1.0684, "L": 1.0679, "H": 1.0691 },
    { "T": 70, "O": 1.0683, "C": 1.0691, "L": 1.068, "H": 1.0692 },
    { "T": 71, "O": 1.069, "C": 1.0701, "L": 1.0689, "H": 1.0703 },
    { "T": 72, "O": 1.0701, "C": 1.0707, "L": 1.0699, "H": 1.0712 },
    { "T": 73, "O": 1.0706, "C": 1.0703, "L": 1.0698, "H": 1.0709 },
    { "T": 74, "O": 1.0703, "C": 1.0707, "L": 1.0692, "H": 1.0715 },
    { "T": 75, "O": 1.0707, "C": 1.0701, "L": 1.0697, "H": 1.0707 },
    { "T": 76, "O": 1.0701, "C": 1.0694, "L": 1.0692, "H": 1.0706 },
    { "T": 77, "O": 1.0693, "C": 1.0698, "L": 1.0692, "H": 1.0701 },
    { "T": 78, "O": 1.0699, "C": 1.0699, "L": 1.0696, "H": 1.0703 },
    { "T": 79, "O": 1.0699, "C": 1.0694, "L": 1.0684, "H": 1.0702 },
    { "T": 80, "O": 1.0694, "C": 1.0715, "L": 1.0687, "H": 1.0719 },
    { "T": 81, "O": 1.0715, "C": 1.0737, "L": 1.0709, "H": 1.0737 },
    { "T": 82, "O": 1.0736, "C": 1.0724, "L": 1.0721, "H": 1.0736 },
    { "T": 83, "O": 1.0724, "C": 1.073, "L": 1.0716, "H": 1.0732 },
    { "T": 84, "O": 1.073, "C": 1.0735, "L": 1.0728, "H": 1.0739 },
    { "T": 85, "O": 1.0734, "C": 1.0717, "L": 1.0717, "H": 1.0745 },
    { "T": 86, "O": 1.0716, "C": 1.0688, "L": 1.0682, "H": 1.0734 },
    { "T": 87, "O": 1.0688, "C": 1.0688, "L": 1.0686, "H": 1.0715 },
    { "T": 88, "O": 1.0689, "C": 1.0673, "L": 1.0663, "H": 1.0698 },
    { "T": 89, "O": 1.0674, "C": 1.0656, "L": 1.0652, "H": 1.0682 },
    { "T": 90, "O": 1.0655, "C": 1.0636, "L": 1.063, "H": 1.0657 },
    { "T": 91, "O": 1.0635, "C": 1.0629, "L": 1.0628, "H": 1.0644 },
    { "T": 92, "O": 1.0628, "C": 1.0627, "L": 1.0623, "H": 1.0635 },
    { "T": 93, "O": 1.0627, "C": 1.0625, "L": 1.062, "H": 1.0627 },
    { "T": 94, "O": 1.0625, "C": 1.0627, "L": 1.0621, "H": 1.063 },
    { "T": 95, "O": 1.0625, "C": 1.0626, "L": 1.0621, "H": 1.0628 },
    { "T": 96, "O": 1.0627, "C": 1.0623, "L": 1.0621, "H": 1.0629 },
    { "T": 97, "O": 1.0623, "C": 1.0615, "L": 1.0605, "H": 1.0629 },
    { "T": 98, "O": 1.0614, "C": 1.0595, "L": 1.0582, "H": 1.0615 },
    { "T": 99, "O": 1.0594, "C": 1.0599, "L": 1.0591, "H": 1.0606 },
    { "T": 100, "O": 1.0598, "C": 1.0604, "L": 1.0595, "H": 1.0604 },
    { "T": 101, "O": 1.0604, "C": 1.0607, "L": 1.0599, "H": 1.061 },
    { "T": 102, "O": 1.0608, "C": 1.0585, "L": 1.0585, "H": 1.0608 },
    { "T": 103, "O": 1.0585, "C": 1.0598, "L": 1.0585, "H": 1.0607 },
    { "T": 104, "O": 1.0596, "C": 1.0604, "L": 1.0585, "H": 1.0612 },
    { "T": 105, "O": 1.0604, "C": 1.0609, "L": 1.0584, "H": 1.0614 },
    { "T": 106, "O": 1.0608, "C": 1.0599, "L": 1.0598, "H": 1.0613 },
    { "T": 107, "O": 1.0598, "C": 1.0621, "L": 1.0598, "H": 1.0625 },
    { "T": 108, "O": 1.062, "C": 1.0628, "L": 1.0614, "H": 1.0629 },
    { "T": 109, "O": 1.0627, "C": 1.0629, "L": 1.0613, "H": 1.0643 },
    { "T": 110, "O": 1.063, "C": 1.061, "L": 1.0604, "H": 1.0636 },
    { "T": 111, "O": 1.0612, "C": 1.0599, "L": 1.0597, "H": 1.0625 },
    { "T": 112, "O": 1.0598, "C": 1.0574, "L": 1.0569, "H": 1.0606 },
    { "T": 113, "O": 1.0575, "C": 1.06, "L": 1.057, "H": 1.06 },
    { "T": 114, "O": 1.0599, "C": 1.0587, "L": 1.0584, "H": 1.0599 },
    { "T": 115, "O": 1.0587, "C": 1.0597, "L": 1.0585, "H": 1.0606 },
    { "T": 116, "O": 1.0598, "C": 1.06, "L": 1.0596, "H": 1.0608 },
    { "T": 117, "O": 1.06, "C": 1.0593, "L": 1.0586, "H": 1.06 },
    { "T": 118, "O": 1.0593, "C": 1.0589, "L": 1.0587, "H": 1.0598 },
    { "T": 119, "O": 1.0593, "C": 1.0587, "L": 1.0587, "H": 1.0595 },
    { "T": 120, "O": 1.0588, "C": 1.0583, "L": 1.0579, "H": 1.0589 },
    { "T": 121, "O": 1.0583, "C": 1.0598, "L": 1.0583, "H": 1.0611 },
    { "T": 122, "O": 1.0599, "C": 1.0606, "L": 1.0596, "H": 1.0614 },
    { "T": 123, "O": 1.0606, "C": 1.0607, "L": 1.0601, "H": 1.0612 },
    { "T": 124, "O": 1.0060, "C": 1.0596, "L": 1.0588, "H": 1.0607 },
    { "T": 125, "O": 1.0596, "C": 1.0595, "L": 1.0591, "H": 1.0599 },
    { "T": 126, "O": 1.0596, "C": 1.0601, "L": 1.0595, "H": 1.0605 },
    { "T": 127, "O": 1.0602, "C": 1.0608, "L": 1.0598, "H": 1.0612 },
    { "T": 128, "O": 1.0608, "C": 1.0631, "L": 1.0604, "H": 1.0636 },
    { "T": 129, "O": 1.0632, "C": 1.0644, "L": 1.0625, "H": 1.0645 },
    { "T": 130, "O": 1.0643, "C": 1.0632, "L": 1.0628, "H": 1.0649 },
    { "T": 131, "O": 1.0631, "C": 1.0631, "L": 1.0619, "H": 1.064 },
    { "T": 132, "O": 1.0631, "C": 1.0633, "L": 1.0618, "H": 1.0637 },
    { "T": 133, "O": 1.0633, "C": 1.0632, "L": 1.0629, "H": 1.0647 },
    { "T": 134, "O": 1.0633, "C": 1.0639, "L": 1.0623, "H": 1.0646 },
    { "T": 135, "O": 1.064, "C": 1.0622, "L": 1.0611, "H": 1.064 },
    { "T": 136, "O": 1.0621, "C": 1.0621, "L": 1.0617, "H": 1.0636 },
    { "T": 137, "O": 1.0622, "C": 1.0597, "L": 1.0597, "H": 1.0629 },
    { "T": 138, "O": 1.0597, "C": 1.0606, "L": 1.0596, "H": 1.0614 },
    { "T": 139, "O": 1.0606, "C": 1.0601, "L": 1.059, "H": 1.0608 },
    { "T": 140, "O": 1.0601, "C": 1.061, "L": 1.06, "H": 1.0612 },
    { "T": 141, "O": 1.061, "C": 1.0625, "L": 1.061, "H": 1.0628 },
    { "T": 142, "O": 1.0625, "C": 1.0629, "L": 1.0625, "H": 1.064 },
    { "T": 143, "O": 1.0631, "C": 1.0631, "L": 1.0625, "H": 1.0636 },
    { "T": 144, "O": 1.0629, "C": 1.0638, "L": 1.0626, "H": 1.0638 },
    { "T": 145, "O": 1.0637, "C": 1.0638, "L": 1.0634, "H": 1.065 },
    { "T": 146, "O": 1.0638, "C": 1.0635, "L": 1.0633, "H": 1.0646 },
    { "T": 147, "O": 1.0635, "C": 1.0629, "L": 1.0627, "H": 1.0638 },
    { "T": 148, "O": 1.0628, "C": 1.0632, "L": 1.0625, "H": 1.0634 },
    { "T": 149, "O": 1.0631, "C": 1.0628, "L": 1.0627, "H": 1.0632 },
    { "T": 150, "O": 1.0627, "C": 1.063, "L": 1.0624, "H": 1.0633 },
    { "T": 151, "O": 1.0631, "C": 1.0615, "L": 1.0615, "H": 1.0633 },
    { "T": 152, "O": 1.0615, "C": 1.0619, "L": 1.0604, "H": 1.0622 },
    { "T": 153, "O": 1.0618, "C": 1.0639, "L": 1.0612, "H": 1.064 },
    { "T": 154, "O": 1.0639, "C": 1.0648, "L": 1.0635, "H": 1.0658 },
    { "T": 155, "O": 1.0647, "C": 1.0633, "L": 1.0629, "H": 1.065 },
    { "T": 156, "O": 1.0632, "C": 1.0631, "L": 1.0627, "H": 1.0644 },
    { "T": 157, "O": 1.063, "C": 1.0611, "L": 1.0604, "H": 1.0631 },
    { "T": 158, "O": 1.0612, "C": 1.0612, "L": 1.0604, "H": 1.0621 },
    { "T": 159, "O": 1.0611, "C": 1.0622, "L": 1.061, "H": 1.0627 },
    { "T": 160, "O": 1.0621, "C": 1.0585, "L": 1.0584, "H": 1.0623 },
    { "T": 161, "O": 1.0585, "C": 1.0618, "L": 1.0585, "H": 1.0625 },
    { "T": 162, "O": 1.0618, "C": 1.0625, "L": 1.0617, "H": 1.0638 },
    { "T": 163, "O": 1.0625, "C": 1.0619, "L": 1.061, "H": 1.0626 },
    { "T": 164, "O": 1.0619, "C": 1.0624, "L": 1.0615, "H": 1.0627 },
    { "T": 165, "O": 1.0623, "C": 1.0625, "L": 1.0619, "H": 1.0629 },
    { "T": 166, "O": 1.0625, "C": 1.0627, "L": 1.0624, "H": 1.063 },
    { "T": 167, "O": 1.0625, "C": 1.0628, "L": 1.0623, "H": 1.0631 },
    { "T": 168, "O": 1.0627, "C": 1.0633, "L": 1.0627, "H": 1.0635 },
    { "T": 169, "O": 1.0633, "C": 1.0625, "L": 1.0625, "H": 1.0635 },
    { "T": 170, "O": 1.0625, "C": 1.0624, "L": 1.0622, "H": 1.0631 },
    { "T": 171, "O": 1.0625, "C": 1.0631, "L": 1.0624, "H": 1.0631 },
    { "T": 172, "O": 1.0631, "C": 1.0626, "L": 1.062, "H": 1.0631 },
    { "T": 173, "O": 1.0627, "C": 1.0626, "L": 1.0621, "H": 1.0627 },
    { "T": 174, "O": 1.0627, "C": 1.063, "L": 1.0626, "H": 1.063 },
    { "T": 175, "O": 1.0629, "C": 1.064, "L": 1.0627, "H": 1.0643 },
    { "T": 176, "O": 1.064, "C": 1.0624, "L": 1.0623, "H": 1.0644 },
    { "T": 177, "O": 1.0624, "C": 1.0609, "L": 1.0606, "H": 1.0627 },
    { "T": 178, "O": 1.0609, "C": 1.0613, "L": 1.06, "H": 1.0617 },
    { "T": 179, "O": 1.0612, "C": 1.0627, "L": 1.0604, "H": 1.0631 },
    { "T": 180, "O": 1.0627, "C": 1.0623, "L": 1.0621, "H": 1.0636 },
    { "T": 181, "O": 1.0622, "C": 1.0612, "L": 1.0612, "H": 1.0623 },
    { "T": 182, "O": 1.0613, "C": 1.0557, "L": 1.0556, "H": 1.0613 },
    { "T": 183, "O": 1.0556, "C": 1.0553, "L": 1.0541, "H": 1.056 },
    { "T": 184, "O": 1.0554, "C": 1.0532, "L": 1.0526, "H": 1.0559 },
    { "T": 185, "O": 1.0535, "C": 1.0559, "L": 1.053, "H": 1.0566 },
    { "T": 186, "O": 1.0559, "C": 1.0557, "L": 1.0549, "H": 1.0566 },
    { "T": 187, "O": 1.0556, "C": 1.0552, "L": 1.0548, "H": 1.0564 },
    { "T": 188, "O": 1.0551, "C": 1.0549, "L": 1.0538, "H": 1.056 },
    { "T": 189, "O": 1.0549, "C": 1.0553, "L": 1.0548, "H": 1.0557 },
    { "T": 190, "O": 1.0553, "C": 1.0554, "L": 1.0549, "H": 1.0554 },
    { "T": 191, "O": 1.0553, "C": 1.0553, "L": 1.0547, "H": 1.0556 },
    { "T": 192, "O": 1.0551, "C": 1.0542, "L": 1.0539, "H": 1.0551 },
    { "T": 193, "O": 1.0543, "C": 1.0539, "L": 1.0533, "H": 1.0545 },
    { "T": 194, "O": 1.0541, "C": 1.0535, "L": 1.0532, "H": 1.0544 },
    { "T": 195, "O": 1.0535, "C": 1.053, "L": 1.0528, "H": 1.0539 },
    { "T": 196, "O": 1.053, "C": 1.0532, "L": 1.0525, "H": 1.0537 },
    { "T": 197, "O": 1.0533, "C": 1.0544, "L": 1.0532, "H": 1.0545 },
    { "T": 198, "O": 1.0543, "C": 1.0547, "L": 1.0541, "H": 1.0551 },
    { "T": 199, "O": 1.0547, "C": 1.054, "L": 1.0539, "H": 1.0547 },
    { "T": 200, "O": 1.0539, "C": 1.0535, "L": 1.052, "H": 1.055 },
    { "T": 201, "O": 1.0537, "C": 1.0555, "L": 1.0518, "H": 1.0568 },
    { "T": 202, "O": 1.0554, "C": 1.0567, "L": 1.0551, "H": 1.0578 },
    { "T": 203, "O": 1.0568, "C": 1.0568, "L": 1.0549, "H": 1.0574 },
    { "T": 204, "O": 1.057, "C": 1.0565, "L": 1.056, "H": 1.0585 },
    { "T": 205, "O": 1.0564, "C": 1.0551, "L": 1.0548, "H": 1.0575 },
    { "T": 206, "O": 1.0551, "C": 1.0563, "L": 1.0543, "H": 1.057 },
    { "T": 207, "O": 1.0563, "C": 1.0573, "L": 1.0556, "H": 1.0574 },
    { "T": 208, "O": 1.0573, "C": 1.0562, "L": 1.0561, "H": 1.0577 },
    { "T": 209, "O": 1.0562, "C": 1.0558, "L": 1.0556, "H": 1.058 },
    { "T": 210, "O": 1.0557, "C": 1.0556, "L": 1.0554, "H": 1.0562 },
    { "T": 211, "O": 1.0555, "C": 1.0555, "L": 1.0549, "H": 1.0558 },
    { "T": 212, "O": 1.0556, "C": 1.0554, "L": 1.0551, "H": 1.0556 },
    { "T": 213, "O": 1.0554, "C": 1.0553, "L": 1.0551, "H": 1.0557 },
    { "T": 214, "O": 1.0553, "C": 1.0554, "L": 1.0552, "H": 1.0556 },
    { "T": 215, "O": 1.0553, "C": 1.0556, "L": 1.0551, "H": 1.0558 },
    { "T": 216, "O": 1.0555, "C": 1.056, "L": 1.0555, "H": 1.0561 }
];

export const dataPivot = [
    { "T": 1483660800, "O": 1.0496, "C": 1.0603, "L": 1.0481, "H": 1.0614 }
];