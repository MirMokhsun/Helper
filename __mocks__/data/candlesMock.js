export const JSONCandle =  [
        { H: 101.76, L: 100.4, O: 100.78, C: 101.03 },
        { H: 101.58, L: 100.27, O: 101.31, C: 101.12 },
        { H: 102.24, L: 100.15, O: 101.41, C: 101.17 },
        { H: 102.28, L: 101.5, O: 102.24, C: 102.23 },
        { H: 102.91, L: 101.78, O: 101.91, C: 102.52 },
        { H: 105.18, L: 103.85, O: 103.96, C: 104.58 },
        { H: 106.31, L: 104.59, O: 104.61, C: 105.97 },
        { H: 106.47, L: 104.96, O: 105.52, C: 105.8 },
        { H: 106.5, L: 105.19, O: 106.34, C: 105.92 },
        { H: 107.65, L: 105.1401, O: 105.93, C: 105.91 },
        { H: 107.29, L: 105.21, O: 105.25, C: 106.72 },
        { H: 107.07, L: 105.9, O: 106.48, C: 106.13 },
        { H: 106.25, L: 104.89, O: 105.47, C: 105.67 },
        { H: 106.19, L: 105.06, O: 106, C: 105.19 },
        { H: 107.79, L: 104.88, O: 104.89, C: 107.7 },
        { H: 110.42, L: 108.6, O: 108.65, C: 109.56 },
        { H: 109.9, L: 108.88, O: 109.72, C: 108.99 },
        { H: 110, L: 108.2, O: 108.78, C: 109.99 },
        { H: 112.19, L: 110.27, O: 110.42, C: 111.08 },
        { H: 110.73, L: 109.42, O: 109.51, C: 109.81 },
        { H: 110.98, L: 109.2, O: 110.23, C: 110.96 },
        { H: 110.42, L: 108.121, O: 109.95, C: 108.54 },
        { H: 109.77, L: 108.17, O: 108.91, C: 108.66 },
        { H: 110.61, L: 108.83, O: 108.97, C: 109.04 },
        { H: 110.5, L: 108.66, O: 109.34, C: 110.44 },
        { H: 112.34, L: 110.8, O: 110.8, C: 112.0192 },
        { H: 112.39, L: 111.33, O: 111.62, C: 112.1 },
        { H: 112.3, L: 109.73, O: 112.11, C: 109.85 },
        { H: 108.95, L: 106.94, O: 108.89, C: 107.48 },
        { H: 108, L: 106.23, O: 107.88, C: 106.91 },
        { H: 108.09, L: 106.06, O: 106.64, C: 107.13 },
        { H: 106.93, L: 105.52, O: 106.93, C: 105.97 },
        { H: 106.48, L: 104.62, O: 105.01, C: 105.68 },
        { H: 105.65, L: 104.51, O: 105, C: 105.08 },
        { H: 105.3, L: 103.91, O: 103.91, C: 104.35 },
        { H: 98.71, L: 95.68, O: 96, C: 97.82 },
        { H: 97.88, L: 94.25, O: 97.61, C: 94.8075 },
        { H: 94.72, L: 92.51, O: 93.99, C: 93.75 },
        { H: 94.08, L: 92.4, O: 93.965, C: 93.65 },
        { H: 95.74, L: 93.68, O: 94.2, C: 95.18 },
        { H: 95.9, L: 93.82, O: 95.2, C: 94.19 },
        { H: 94.07, L: 92.68, O: 94, C: 93.24 },
        { H: 93.45, L: 91.85, O: 93.37, C: 92.72 },
        { H: 93.77, L: 92.59, O: 93, C: 92.82 },
        { H: 93.57, L: 92.11, O: 93.33, C: 93.39 },
        { H: 93.57, L: 92.46, O: 93.48, C: 92.51 },
        { H: 92.78, L: 89.47, O: 92.72, C: 90.32 },
        { H: 91.67, L: 90, O: 90, C: 90.52 }
    ]


export const JSONCandle2 = [
        { y: 101.03 },
        { y: 101.12 },
        { y: 101.17 },
        { y: 102.23 },
        { y: 102.52 },
        { y: 104.58 },
        { y: 105.97 },
        { y: 105.8 },
        { y: 105.92 },
        { y: 105.91 },
        { y: 106.72 },
        { y: 106.13 },
        { y: 105.67 },
        { y: 105.19 },
        { y: 107.7 },
        { y: 109.56 },
        { y: 108.99 },
        { y: 109.99 },
        { y: 111.08 },
        { y: 109.81 },
        { y: 110.96 },
        { y: 108.54 },
        { y: 108.66 },
        { y: 109.04 },
        { y: 110.44 },
        { y: 112.0192 },
        { y: 112.1 },
        { y: 109.85 },
        { y: 107.48 },
        { y: 106.91 },
        { y: 107.13 },
        { y: 105.97 },
        { y: 105.68 },
        { y: 105.08 },
        { y: 104.35 },
        { y: 102 },
        { y: 104.8075 },
        { y: 103.75 },
        { y: 103.65 },
        { y: 100.18 },
        { y: 119 },
        { y: 104 },
        { y: 102.72 },
        { y: 102 },
        { y: 103.39 },
        { y: 102.51 },
        { y: 100.32 },
        { y: 90.52 }
]

// export const JSONCandle2 = {...candle2};