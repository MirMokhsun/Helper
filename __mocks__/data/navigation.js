export const navigation = {
    navigate: () => { },
    setParams: () => { },
    state: { params: 'test' },
    addListener: () => { },
}