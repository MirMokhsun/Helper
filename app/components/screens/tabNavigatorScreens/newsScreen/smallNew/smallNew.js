import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { View, Text, TouchableOpacity } from 'react-native';
import SmallClockGray from './svg/smallClockGray';
import { withNavigation } from 'react-navigation';
import { setNewsDetails } from "./redux/setNewsDetailsAction";
import store from '../../../../../redux/store/store';
import { styles } from './styles';
import { localization } from '../../../../../common/localization';

export class SmallNew extends PureComponent {
    constructor(props) {
        super(props);
    }

    goToNewsDetails = () => {
        const { navigation, id, setNewsDetails, scrollView, randomNumb, language } = this.props;
        scrollView.scrollTo({ x: 0, y: 0, animated: false });
        navigation.navigate('NewsDetailsScreen', { randomNumb, title:  localization[language].News });
        store.dispatch(setNewsDetails(id));
    }

    render() {
        const { image, textNews, timeNews, id, currentTextAlign } = this.props;
        return (
            <TouchableOpacity onPress={this.goToNewsDetails} testID={`SmallNew_${id}ID`} accessibilityLabel={`SmallNew_${id}ID`}>
                <View style={styles.maincontainer} alignSelf="center">
                    <View style={styles.imageContainner}>
                        {image}
                    </View>
                    <Text style={[styles.textNewsStyle, { textAlign: currentTextAlign || 'left' }]}>{textNews}</Text>
                    <View style={styles.timeNewsWrapper}>
                        <SmallClockGray />
                        <Text style={styles.timeNewsStyle}>{timeNews}</Text>
                    </View>
                </View>
            </TouchableOpacity >
        );
    }
}
const mapStateToProps = state => ({
    language: state.currentLanguage.language,
    navigation: state.stackNavigator.navigator
});
const mapDispatchToProps = dispatch => bindActionCreators({
    setNewsDetails
}, dispatch);
export default withNavigation(connect(mapStateToProps, mapDispatchToProps)(SmallNew));