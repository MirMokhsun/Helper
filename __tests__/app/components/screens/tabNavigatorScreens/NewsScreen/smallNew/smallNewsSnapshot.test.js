import { Image } from 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { SmallNew } from '../../../../../../../app/components/screens/tabNavigatorScreens/NewsScreen/smallNew/smallNew';
import { navigation } from '../../../../../../../__mocks__/data/navigation';

it('create snapshot SmallNew', () => {
    const image = <Image source={require("../../../../../../../app/components/screens/tabNavigatorScreens/newsScreen/images/defaultClosedImages/1.jpg")} resizeMode={"stretch"} style={{}} />;
    const textNews = "cutTextNews";
    const timeNews = "10:27 12 октября, 2018";

    const renderer = new ShallowRenderer();
    const smallNew = renderer.render(<SmallNew image={null}
        textNews={textNews}
        timeNews={timeNews}
        id={1}
        randomNumb={1}
        navigation={navigation}
        currentTextAlign={'auto'}
        setNewsDetails={() => { }}
        scrollView={scrollTo => { }}
    />);
    expect(smallNew).toMatchSnapshot();
});