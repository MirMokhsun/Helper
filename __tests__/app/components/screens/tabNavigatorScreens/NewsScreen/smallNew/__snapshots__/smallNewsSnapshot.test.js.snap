// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot SmallNew 1`] = `
<TouchableOpacity
  accessibilityLabel="SmallNew_1ID"
  activeOpacity={0.2}
  onPress={[Function]}
  testID="SmallNew_1ID"
>
  <View
    alignSelf="center"
    style={
      Object {
        "marginBottom": 20,
        "width": 730,
      }
    }
  >
    <View
      style={
        Object {
          "height": 80,
          "overflow": "hidden",
          "position": "relative",
          "zIndex": -2,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000756",
            "fontFamily": "Roboto-Light",
            "fontSize": 16,
            "lineHeight": 20,
            "marginBottom": 10,
            "marginTop": 10,
          },
          Object {
            "textAlign": "auto",
          },
        ]
      }
    >
      cutTextNews
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <SmallClockGray />
      <Text
        style={
          Object {
            "color": "#1459D2",
            "fontFamily": "Roboto-Light",
            "fontSize": 11,
            "lineHeight": 13,
          }
        }
      >
        10:27 12 октября, 2018
      </Text>
    </View>
  </View>
</TouchableOpacity>
`;
